common/PerfStressShared/PerfStressUtilities.cs

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using CommandLine;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Emit;
+using System.Text.RegularExpressions;
+using System.Threading;
+
+namespace Azure.Test.PerfStress
+{
+    internal static class PerfStressUtilities
+    {
+        private static readonly Lazy<Parser> _parser = new Lazy<Parser>(() => new Parser(settings =>
+        {
+            settings.CaseSensitive = false;
+            settings.HelpWriter = Console.Error;
+        }));
+
+        internal static Parser Parser => _parser.Value;
+
+        internal static bool ContainsOperationCanceledException(Exception e)
+        {
+            if (e is OperationCanceledException)
+            {
+                return true;
+            }
+            else if (e.InnerException != null)
+            {
+                return ContainsOperationCanceledException(e.InnerException);
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        // Dynamically create option types with a "Verb" attribute
+        internal static Type[] GetOptionTypes(IEnumerable<Type> testTypes)
+        {
+            var optionTypes = new List<Type>();
+
+            var ab = AssemblyBuilder.DefineDynamicAssembly(new AssemblyName("Options"), AssemblyBuilderAccess.Run);
+            var mb = ab.DefineDynamicModule("Options");
+
+            foreach (var t in testTypes)
+            {
+                var baseOptionsType = t.GetConstructors().First().GetParameters()[0].ParameterType;
+                var tb = mb.DefineType(t.Name + "Options", TypeAttributes.Public, baseOptionsType);
+
+                var attrCtor = typeof(VerbAttribute).GetConstructor(new Type[] { typeof(string), typeof(bool) });
+                var verbName = GetVerbName(t.Name);
+                tb.SetCustomAttribute(new CustomAttributeBuilder(attrCtor,
+                    new object[] { verbName, attrCtor.GetParameters()[1].DefaultValue }));
+
+                optionTypes.Add(tb.CreateType());
+            }
+
+            return optionTypes.ToArray();
+        }
+
+        internal static string GetVerbName(string testName)
+        {
+            // Remove suffix "Test" if exists
+            return Regex.Replace(testName, "Test$", string.Empty, RegexOptions.IgnoreCase);
+        }
+
+        // Run in dedicated thread instead of using async/await in ThreadPool, to ensure this thread has priority
```

Comment:
1. Why does a status printing thread need priority? Seems test runners should have priority over that.
2. Considering that, might set `IsBackground` to avoid this keeping the program alive if not properly cancelled or killed.

Reply:
The status printing thread should be very inexpensive, but it can be starved when testing a very fast scenario like a no-op method.  I think it's better to ensure the status printing always works, even if it slightly reduces performance of very fast scenarios.  It should not matter either way for real-world tests.

If there is a better way to ensure the status printing never gets starved, other than running in a dedicated thread, I'm happy to change it.

I'd rather not set `IsBackground`, since the code already ensures the status printing thread is killed before printing the final results.  If the thread was not killed it would interfere with the final results.